src/MorseL/Extensions/MorseLExtensions.cs

```diff
@@ -44,9 +44,9 @@ public static IMorseLBuilder AddMorseL(this IServiceCollection services, Action<
         /// <param name="builder">The MorseLBuilder object</param>
         /// <param name="lifetime">The lifetime the middleware should be created and adhere to.</param>
         /// <returns></returns>
-        public static IMorseLBuilder AddMiddleware<TMiddleware>(this IMorseLBuilder builder, ServiceLifetime lifetime) where TMiddleware : IMiddleware
+        public static IMorseLBuilder AddMiddleware<TMiddleware>(this IMorseLBuilder builder, ServiceLifetime lifetime) where TMiddleware : Sockets.Middleware.IMiddleware
```

Comment:
Was there another IMiddleware introduced somewhere to need to fully qualify the name here?

Reply:
Yes, this became an ambiguous reference with .Net Core 2.0. Microsoft added an IMiddleware in one of the MS namespaces that were referenced in this class.